### PR TITLE
[Spark-339][WIP] golang cli

### PR DESCRIPTION
**Pull request by ArtRand**
Now:
- [x] Change `principal` to `service-account`
- [ ] Vendor dependencies 
~- [ ] Add test for "properties file"~

After making new Spark distribution:
- [x] Add generic secret support, enable test
~- [ ] Update and align documentation~

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -18,8 +18,8 @@
 # docker build -t spark:git-`git rev-parse --short HEAD` .
 
 # Basing from Mesos image so the Mesos native library is present.
```
Comment (elezar) [inline docker/Dockerfile]: This comment is no longer valid.

**Comment by ArtRand** [comment]
retest this please

**Comment by ArtRand** [comment]
restest this please (ccm)

**Comment by ArtRand** [comment]
retest this please (hdfs)

File: cli/dcos-spark/submit_builder.go [hunk 2]
```diff
@@ -146,6 +147,10 @@ Args:
 	submit.Flag("tgt-secret-value", "Value of TGT to be used in the drivers, must be base64 encoded").
 		Default("").StringVar(&args.tgtSecretValue)
 
+	// TODO expose krb5 config (base64)
```
Comment (nickbp) [inline cli/dcos-spark/submit_builder.go]: probably can't hurt to update the docs at L78-L116 to reflect the current state of things

File: bin/universe.sh [hunk 3]
```diff
@@ -55,28 +58,37 @@ fetch_commons_tools() {
...
     TEMPLATE_DOCKER_IMAGE=${DOCKER_IMAGE} \
         ${COMMONS_TOOLS_DIR}/ci_upload.py \
             spark \
-            ${DISPATCHER_DIR}/package/ \
-            ${DISPATCHER_DIR}/cli/dist/*.whl
+            ${SPARK_BUILD_DIR}/universe/ \
+            ${SPARK_BUILD_DIR}/cli/dcos-spark/dcos-spark-darwin \
+            ${SPARK_BUILD_DIR}/cli/dcos-spark/dcos-spark-linux \
+            ${SPARK_BUILD_DIR}/cli/dcos-spark/dcos-spark.exe \
+            ${SPARK_BUILD_DIR}/cli/python/dist/*.whl
```
Comment (nickbp) [inline bin/universe.sh]: Does this release of spark need to support DC/OS 1.7? If not, then the whl python wrapper (and everything in cli/python/) can be removed . 1.8 was when native binary CLI modules was introduced; the python wrapper was effectively a 1.7 compatibility shim.
- Remove this `*.whl` entry
- `rm -rf cli/python/`
- Remove python wrapper build call in cli/Makefile
- `rm universe/command.json` (or wherever the universe packaging is in the repo)

File: manifest.json [hunk 4]
```diff
@@ -3,16 +3,16 @@
     "spark_version": "2.2.0",
     "default_spark_dist": {
         "hadoop_version": "2.6",
-        "uri": "https://downloads.mesosphere.com/spark/assets/spark-2.2.0-bin-2.6.tgz"
+        "uri": "https://s3-us-west-2.amazonaws.com/arand-sandbox-mesosphere/autodelete7d/spark-release-candidate-bin-2.6.tgz"
```
Comment (nickbp) [inline manifest.json]: don't forget to clean these up..

File: bin/universe.sh (hunk 3 above)
Comment (nickbp) [inline bin/universe.sh]: Actually, per offline discussion, it'd make sense to keep the python wrapper for now as it's needed for compatibility with spark-build's old tooling that's still on universe-2.x.

File: manifest.json (hunk 4 above)
Comment (ArtRand) [inline manifest.json]: yeah, this is just to test other builds with jenkins

**Comment by ArtRand** [comment]
retest this please (hdfs timeout?)

File: tests/utils.py [hunk 5]
```diff
@@ -187,14 +187,14 @@ def upload_file(file_path):
...
-def submit_job(app_url, app_args, args=[]):
+def submit_job(app_url, app_args, app_name="/spark", args=[]):
     if is_strict():
         args += ["--conf", 'spark.mesos.driverEnv.MESOS_MODULES=file:///opt/mesosphere/etc/mesos-scheduler-modules/dcos_authenticatee_module.json']
         args += ["--conf", 'spark.mesos.driverEnv.MESOS_AUTHENTICATEE=com_mesosphere_dcos_ClassicRPCAuthenticatee']
         args += ["--conf", 'spark.mesos.principal=service-acct']
     args_str = ' '.join(args + ["--conf", "spark.driver.memory=2g"])
     submit_args = ' '.join([args_str, app_url, app_args])
-    cmd = 'dcos --log-level=DEBUG spark --verbose run --submit-args="{0}"'.format(submit_args)
+    cmd = 'dcos spark --name={app_name} --verbose run --submit-args="{args}"'.format(app_name=app_name, args=submit_args)
```
Comment (susanxhuynh) [inline tests/utils.py]: You didn't want to keep the DEBUG log level?

File: universe/package.json [hunk 6]
```diff
@@ -1,11 +1,11 @@
 {
     "packagingVersion": "3.0",
-    "postInstallNotes": "DC/OS Spark is being installed!\n\n\tDocumentation: https://docs.mesosphere.com/service-docs/spark/\n\tIssues: https://docs.mesosphere.com/support/",
+    "postInstallNotes": "DC/OS Spark is being installed!\n\n\tDocumentation: https://docs.mesosphere.com/current/usage/service-guides/spark/\n\tIssues: https://docs.mesosphere.com/support/",
```
Comment (susanxhuynh) [inline universe/package.json]: I don't think the docs URL needs to be changed here. Same for "postUninstallNotes", "description", and "website" below.

File: universe/package.json [hunk 7]
```diff
@@ -18,8 +18,8 @@
         "batch",
         "analytics"
     ],
-    "website": "https://docs.mesosphere.com/service-docs/spark/",
+    "website": "https://docs.mesosphere.com/current/usage/service-guides/spark/",
     "selected": true,
     "version": "{{package-version}}",
-    "minDcosReleaseVersion": "1.8"
+    "minDcosReleaseVersion": "1.7"
```
Comment (susanxhuynh) [inline universe/package.json]: Should be 1.8?

File: tests/utils.py (hunk 5 above)
Comment (ArtRand) [inline tests/utils.py]: The new CLI doesn't need this flag, `--verbose` does the same thing.

File: bin/universe.sh [hunk 8]
```diff
@@ -15,18 +15,20 @@
 
 set -e -x -o pipefail
 
+
+echo "starting universe.sh"
```
Comment (ArtRand) [inline bin/universe.sh]: Yes, basically.

**Comment by ArtRand** [comment]
retest this please (dcos ImportError?)

File: cli/tests/integration/test_run.py [hunk 9]
```diff
@@ -2,13 +2,8 @@
 from dcos_spark import spark_submit
 
 import mock
-import pytest
 
 
-@pytest.mark.skip(
-    reason=("This is failing when fake.com fails to connect, "
-            "but I don't want to spend the time to fix it, because this CLI "
-            "is hopefully going away in 1.10."))
```
Comment (ArtRand) [inline cli/tests/integration/test_run.py]: the new `test_supervise()` does this now

**Comment by ArtRand** [comment]
retest this please (hfs)

**Comment by ArtRand** [comment]
retest this please (ccm)

File: bin/dcos-commons-tools/airgap_linter.py [hunk 10]
```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
```
Comment (ArtRand) [inline bin/dcos-commons-tools/airgap_linter.py]: Yes, I agree. This was partially to fix the recent changes in dcos-commons and it's also consistent with the rest of the standalone repos.

**Comment by ArtRand** [comment]
retest this please (ccm)

**Comment by ArtRand** [comment]
retest this please

**Pull request merged**